src/protocol/sender/mod.rs

```diff
@@ -0,0 +1,499 @@
+use std::collections::VecDeque;
+use std::net::SocketAddr;
+use std::time::{Duration, Instant};
+
+use bytes::Bytes;
+use log::debug;
+use log::{trace, warn};
+
+use crate::loss_compression::decompress_loss_list;
+use crate::packet::{ControlTypes, HandshakeControlInfo, SrtControlPacket};
+use crate::protocol::handshake::Handshake;
+use crate::protocol::sender::buffers::*;
+use crate::protocol::Timer;
+use crate::Packet::*;
+use crate::{
+    CCData, CongestCtrl, ConnectionSettings, ControlPacket, DataPacket, Packet, SeqNumber,
+    SrtCongestCtrl, Stats,
+};
+
+pub mod buffers;
+
+#[derive(Debug)]
+pub enum SenderError {}
+
+pub type SenderResult = Result<(), SenderError>;
+
+pub struct SenderMetrics {
```

Comment:
This seems very similar to `Stats`, there should only be one. I prefer the name `SenderMetrics`, considering receiver doesn't currently expose `Stats`.